android/source/VideoLocker/src/org/edx/mobile/view/VideoListActivity.java

```diff
@@ -125,7 +125,15 @@ protected void onSaveInstanceState(Bundle outState) {
     @Override
     protected void onRestoreInstanceState(Bundle savedInstanceState) {
         if (savedInstanceState != null) {
-            super.onRestoreInstanceState(savedInstanceState);
+            //FIXME: There is an error of null pointer exception on some devices when super function is called.
+            //This is a workaround where the crash is handled in
+            //try catch and hence the app does not crash
+            try{
+                super.onRestoreInstanceState(savedInstanceState);
+            }catch(Exception e){
+                e.printStackTrace();
+                LogUtil.error("player_issue","Null pointer exception on onRestoreInstanceState");
```

Comment:
Instead of calling `printStackTrace` before each error message, you should change the `LogUtil.error` method to always log the stack trace.  The built-in logging facilities do this by calling the method that takes `Throwable`:
http://developer.android.com/reference/android/util/Log.html#e(java.lang.String,%20java.lang.String,%20java.lang.Throwable)

As mentioned earlier, the advantage of using the built-in logging library is that these statements can be automatically disabled/enabled depending on the build configuration and not having print statements dispersed throughout the codebase.